iroh-bytes/src/store/traits.rs

```diff
@@ -103,23 +110,178 @@ pub trait Map: Clone + Send + Sync + 'static {
 
 /// A partial entry
 pub trait PartialMapEntry<D: PartialMap>: MapEntry<D> {
-    /// A future that resolves to an writeable outboard
-    fn outboard_mut(&self) -> BoxFuture<'_, io::Result<D::OutboardMut>>;
-    /// A future that resolves to a writer that can be used to write the data
-    fn data_writer(&self) -> BoxFuture<'_, io::Result<D::DataWriter>>;
+    /// Get a batch writer
+    fn batch_writer(&self) -> BoxFuture<'_, io::Result<D::BatchWriter>>;
+}
+
+/// An async batch interface for writing bao content items to a pair of data and
+/// outboard.
+///
+/// Details like the chunk group size and the actual storage location are left
+/// to the implementation.
+pub trait BaoBatchWriter {
+    /// Write a batch of bao content items to the underlying storage.
+    ///
+    /// The batch is guaranteed to be sorted as data is received from the network.
+    /// So leafs will be sorted by offset, and parents will be sorted by pre order
+    /// traversal offset. There is no guarantee that they will be consecutive
+    /// though.
+    ///
+    /// The size is the size the remote side told us. It is not guaranteed
```

Comment:
this is the total size of all batch items?

Reply:
No, this is the size in bytes of the entire blob. We just can't fully trust it yet. But in the previous impl we did trust it.

This determines what the stuff on the wire means, so this is an *upper bound* on the size, so are guaranteed to only get stuff within 0..size.

Reply:
Trying to clarify this more in the comments.

Reply:
maybe call it blob_size or sth like that